DependencyInjection/Compiler/MiddlewaresPass.php

```diff
@@ -54,6 +66,18 @@ public function process(ContainerBuilder $container): void
                 $childDef->addMethodCall('setConnectionName', [$name]);
             }
 
+            $middlewareDefs = array_map(
+                static fn ($id, $def) => [
+                    $middlewareConnections[$id][$name] ?? $middlewarePriorities[$id] ?? 0,
+                    $def[1],
+                    $def[0],
+                ],
+                array_keys($middlewareDefs),
+                array_values($middlewareDefs),
+            );
+            uasort($middlewareDefs, static fn ($a, $b) => $b[0] <=> $a[0] ?: $a[1] <=> $b[1]);
```

Comment:
maybe we can use https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php#L40

Reply:
It was my first idea, but it doesn't work :(. On [line 56](https://github.com/symfony/symfony/blob/52a92926f7fed15cdff399c6921100a10e0d6f61/src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php#L56), `PriorityTaggedServiceTrait` calls `$container->findTaggedServiceIds` with `throwOnAbstract: true`, which is incompatible with Doctrine Middleware that can be abstract definitions (as it is the case in the tests).

Reply:
Ok I see :confused: 

But from what I can see you made sure we are doing a stable sort, right? So the order for null priority tagged services remains the same as before this change?

Reply:
Yes, exactly. In case of identical priority (or no priority set), the order from registration in the container is preserved.